### ISSUE TITLE
RTSP source stop stream after 2 minutes

### ISSUE THREAD

**Issue by Forest-fox** [aler9/mediamtx#1739]
## Which version are you using?

v0.22.2

## Which operating system are you using?

- [x ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

I have setup rtsp proxy with one source - h264 video + g711U audio
I can take source directly with rtsp://ip/av0_0
This link setup as source for mediamtx path

If i play stream directly with SMPlayer - all fine, stream plays for a moment, when i stopped it.
But if i play by proxy - got this:
- stream plays, all fine
- after ~2minutes stream suddenly stopped with messages in log:
2023/04/26 16:59:30 INF [path 10.154.249.2] [rtsp source] stopped
2023/04/26 16:59:30 INF [RTSP] [conn 127.0.0.1:52970] closed (terminated)
2023/04/26 16:59:30 INF [RTSP] [session 381853316] destroyed (terminated)

## Describe how to replicate the issue

1. start the server
2. publish with rtsp
3. read with rtsp

## Did you attach the server logs?

yes

## Did you attach a network dump?

yes

**Comment by aler9** [comment]
Hello @Forest-fox, i'm trying to debug the logs you provided but i think you're reporting two separate issues:

* the initial log you posted reports `[path 10.154.249.2] [rtsp source] stopped` - probably it has to do with `sourceOnDemand: yes`
* the second log you posted reports `[rtsp source] ERR: EOF` - probably it's issue #1812

**Comment by Forest-fox** [comment]
Yes, issue is second variant with [rtsp source] ERR: EOF / muxer destroyed (terminated)

**Comment by Forest-fox** [comment]
Have checked version 0.23.2 - also affected
Log, configuration and network dump - [0.23.2-logs.tar.gz](https://drive.google.com/file/d/1iPfu1gofV15UWsgglRG4E-ijyTRnpDqY/view?usp=sharing)

**Comment by aler9** [comment]
From the logs it seems that the camera closes the TCP connection after exactly 2 minutes and 15 seconds. This may be linked to the ability to keep a RTSP session running even when its TCP connection gets closed, which is currently missing (https://github.com/bluenviron/gortsplib/issues/162).

If this hypothesis is right, you might be able to temporarily fix the issue by setting `sourceProtocol: tcp` int he configuration file.

**Comment by Forest-fox** [comment]
Tried to set `sourceProtocol: tcp` for path - not working.  Same behavior.

**Comment by aler9** [comment]
Can you post a network dump of 3 minutes of communication between SMPlayer and the camera? this is needed in order to compare a working setup with the server behavior.

**Comment by Forest-fox** [comment]
https://drive.google.com/file/d/1XfkMkmdy8Lqr7DzwTkIi08JcpVVL7B2M/view?usp=sharing

**Comment by aler9** [comment]
Hello, from the logs you posted it seems that the difference between MediaMTX and SMPlayer is that MediamTX doesn't send RTCP receiver reports to the camera, because the camera doesn't send RTCP sender reports, while SMPlayer, which is GStreamer-based, sends them anyway without waiting for sender reports.

I've created a fix (https://github.com/bluenviron/gortsplib/pull/318) but before merging it i need you to test the following nightly release in order to check whether the issue has been solved.

[mediamtx_v0.23.6-4-gf25754b_darwin_amd64.tar.gz](https://github.com/bluenviron/mediamtx/files/11856695/mediamtx_v0.23.6-4-gf25754b_darwin_amd64.tar.gz)
[mediamtx_v0.23.6-4-gf25754b_darwin_arm64.tar.gz](https://github.com/bluenviron/mediamtx/files/11856696/mediamtx_v0.23.6-4-gf25754b_darwin_arm64.tar.gz)
[mediamtx_v0.23.6-4-gf25754b_linux_amd64.tar.gz](https://github.com/bluenviron/mediamtx/files/11856697/mediamtx_v0.23.6-4-gf25754b_linux_amd64.tar.gz)
[mediamtx_v0.23.6-4-gf25754b_linux_arm64v8.tar.gz](https://github.com/bluenviron/mediamtx/files/11856698/mediamtx_v0.23.6-4-gf25754b_linux_arm64v8.tar.gz)
[mediamtx_v0.23.6-4-gf25754b_linux_armv6.tar.gz](https://github.com/bluenviron/mediamtx/files/11856699/mediamtx_v0.23.6-4-gf25754b_linux_armv6.tar.gz)
[mediamtx_v0.23.6-4-gf25754b_linux_armv7.tar.gz](https://github.com/bluenviron/mediamtx/files/11856700/mediamtx_v0.23.6-4-gf25754b_linux_armv7.tar.gz)
[mediamtx_v0.23.6-4-gf25754b_windows_amd64.zip](https://github.com/bluenviron/mediamtx/files/11856701/mediamtx_v0.23.6-4-gf25754b_windows_amd64.zip)

**Comment by Forest-fox** [comment]
Thanks! I'll test at Monday and give feedback.

**Comment by Forest-fox** [comment]
I can confirm, that problem resolved with fixed version. 
None errors. Stream works fine for more than 10 minutes.

**Comment by aler9** [comment]
added in v0.23.7
